File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.java
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        HashMap<Integer,Integer> map = new HashMap<>();
        for(int i : nums1){
            map.put(i, map.getOrDefault(i , 0)+1);
        }
        ArrayList<Integer> arr = new ArrayList<>();
        for(int i = 0 ; i < nums2.length ; i++ ){
           if(map.containsKey(nums2[i]) && map.get(nums2[i])>0){
               map.put(nums2[i],map.get(nums2[i])-1);
               arr.add(nums2[i]);
           }
        }
        int [] a = new int [arr.size()];
        for(int i = 0 ; i < arr.size() ; i++ ){
            a[i] = arr.get(i);
        }
        return a;
    }
}

File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int,int> mp1;
        unordered_map<int,int> mp2;
        vector<int> ans;
        for(auto n:nums1){
            mp1[n]++;
        }
        for(auto n:nums2){
            mp2[n]++;
        }

        for(auto m:mp1){
            auto found = mp2.find(m.first);
            if(found!=mp2.end()){
                int r = min(found->second, m.second);
                // if(m.second==)
                for(int i=0; i<r; i++){
                    ans.push_back(m.first);
                }
            }
        }
        return ans;

    }
};